clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -668,6 +668,8 @@ struct MCDCCoverageBuilder {
   llvm::SmallVector<const BinaryOperator *> NestLevel;
   llvm::DenseMap<const Stmt *, MCDCConditionID> &CondIDs;
   llvm::DenseMap<const Stmt *, unsigned> &MCDCBitmapMap;
+  llvm::DenseMap<const Stmt *, MCDCConditionID> TrueCondIDs;
```

Comment:
We can use these to track a Condition's True/False IDs and make all of this self-contained within MCDCBuilder rather than calculating it in the visitor routines. Much cleaner.